src/uproot/writing/identify.py

```diff
@@ -659,6 +660,34 @@ def _fXbins_maybe_regular(axis, boost_histogram):
             return axis.edges
 
 
+def _fLabels_maybe_categorical(axis, boost_histogram):
+    if boost_histogram is None:
+        return None
+
+    if not isinstance(
+        axis, (boost_histogram.axis.IntCategory, boost_histogram.axis.StrCategory)
+    ):
+        return None
+
+    labels = [str(label) for label in axis]
+    if isinstance(axis, boost_histogram.axis.IntCategory):
+        # Check labels are valid integers (this may be redundant)
+        for label in labels:
+            try:
+                int(label)
+            except ValueError:
+                raise ValueError(
+                    f"IntCategory labels must be valid integers. Found {label} on axis {axis}"
+                )
```

Comment:
You're right; this might be redundant with https://github.com/scikit-hep/uproot5/pull/764/files#r1006165021

pre-commit was complaining about not specifying whether you want exception-chaining on the second ValueError. Although I said we default to exception-chaining, this is such a short-distance case that chaining would be redundant. I'd do `from None` here:

```suggestion
        # Check labels are valid integers (this may be redundant)
        for label in labels:
            try:
                int(label)
            except ValueError:
                raise ValueError(
                    f"IntCategory labels must be valid integers. Found {label!r} on axis {axis!r}"
                ) from None
```

I'm also adding `!r` to the error message so that it prints their `repr` strings, rather than their `str` strings. The `label` will be quoted, which is good for an error message. (One of us, debugging: "Are the spaces included in `label` or am I confusing that with the rest of the error message?")